independent-projects/tools/platform-descriptor-resolver-json/src/main/java/io/quarkus/platform/descriptor/resolver/json/QuarkusJsonPlatformDescriptorResolver.java

```diff
@@ -278,14 +278,15 @@ private QuarkusPlatformDescriptor loadFromJsonArtifact(AppModelResolver artifact
         try {
             log.debug("Attempting to resolve Quarkus JSON platform descriptor as %s", jsonArtifact);
             return loadFromFile(artifactResolver.resolve(jsonArtifact));
-        } catch (Throwable e) {
+        } catch (Exception e) {
```

Comment:
> change use of Throwable to Exception as if Throwable happens we have worse problems.

For the current state of the code, changing it to Exception is going to break it. IllegalStateException is currently thrown from every method. So until proper checked exceptions are in place, don't change this.

Reply:
IllegalStateException is an Exception.

The only things that can throw descendents of Throwables that are not Exceptions are things like OutOfMemory or worse.  If that happens you want it to surface not swallow it.